pyrogram/client/client.py

```diff
@@ -1123,6 +1123,7 @@ def save_file(self,
                   progress_args: tuple = ()):
         part_size = 512 * 1024
         file_size = os.path.getsize(path)
+        assert file_size < 1500 * 1024 * 1024, "Couldn't upload file bigger 1500MiB"
```

Comment:
- Not a fan of `assert`s in these scenarios. Raising `ValueError` is preferred.
- Files of exactly 1500 MiB are supported, so you wanted `<=` instead.
- I also like more descriptive error messages.

Suggestion:

```suggestion
        if file_size > 1500 * 1024 * 1024:
            raise ValueError("Telegram doesn't support uploading files bigger than 1500 MiB")
```